### ISSUE TITLE
park keeps crashing... Help!

### ISSUE THREAD

**Issue by scottnewm** [OpenRCT2/OpenRCT2#7700]
**OS:** MacOS
**Version:** 0.2.0
**Commit/Build:** 26666b1

im currently playing a park i download and just recently it has been crashing after a few mins of playing. the only way to make it stop crashing is to remove all guests from the park. Also if there are a bunch of guest in the park i have to keep the game paused. What should i do? I love the park and don't want to loose it. Thank you for your help!

- [ ] Reproducible in RCT2 (vanilla) no it won't load there. never has
- [ ] Specific to multiplayer? no

**Steps to reproduce:**
1. 
2. 

**Dump file**
 i have a DAT folder i attached

**Screenshots / Video:**

**Save game:**

[Six Flags Great America 2016 S01.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/2107805/Six.Flags.Great.America.2016.S01.sv6.zip)

[DAT.zip](https://github.com/OpenRCT2/OpenRCT2/files/2107806/DAT.zip)

**Comment by janisozaur** [comment]
Crash happens at 0025-05-04

https://github.com/OpenRCT2/OpenRCT2/blob/13224104e49c9f8b24070ace3e5fe87402b89ad3/src/openrct2/peep/Guest.cpp#L4101

```
>>> p vehicle_entry->peep_loading_positions
$3 = std::vector of length 0, capacity 0
```

Stack trace:
```
>>> bt full
#0  0x00007ffff6f3a94c in rct_peep::UpdateRideLeaveVehicle (this=0x7ffff7cd1ac0 <_spriteList+1618176>) at ../src/openrct2/peep/Guest.cpp:4101
        exitLocation = {
          x = 174, 
          y = 135, 
          z = 14, 
          direction = 0 '\000'
        }
        platformLocation = {
          x = 5532, 
          y = 4304, 
          z = 14, 
          direction = 2 '\002'
        }
        loadPosition = -1 '\377'
        ride = 0x7ffff78b8c90 <gRideList+96368>
        vehicle = 0x7ffff7cf94c0 <_spriteList+1780480>
        ride_station = 0 '\000'
        rideEntry = 0x7fffb5823a98
        vehicle_entry = 0x7fffb5823ab2
        __PRETTY_FUNCTION__ = "void rct_peep::UpdateRideLeaveVehicle()"
        exitLocation = {
          x = 5532, 
          y = 4304, 
          z = 14, 
          direction = 2 '\002'
        }
        waypointLoc = {
          x = 32767, 
          y = -151737860, 
          z = 32767
        }
        trackElement = 0x0
        station_direction = 127 '\177'
        vehicleEntry = 0x555555614040 <_start>
        exitWaypointLoc = {
          x = 174, 
          y = 135, 
          z = 14
        }
#1  0x00007ffff6f3d232 in rct_peep::UpdateRide (this=0x7ffff7cd1ac0 <_spriteList+1618176>) at ../src/openrct2/peep/Guest.cpp:5073
        __PRETTY_FUNCTION__ = "void rct_peep::UpdateRide()"
#2  0x00007ffff6f46cb5 in rct_peep::Update (this=0x7ffff7cd1ac0 <_spriteList+1618176>) at ../src/openrct2/peep/Peep.cpp:1266
        stepsToTake = 128
        carryCheck = 282
        __PRETTY_FUNCTION__ = "void rct_peep::Update()"
#3  0x00007ffff6f44eef in peep_update_all () at ../src/openrct2/peep/Peep.cpp:440
        i = 210
        spriteIndex = 8780
        peep = 0x7ffff7cd1ac0 <_spriteList+1618176>
#4  0x00007ffff6e08ae1 in OpenRCT2::GameState::UpdateLogic (this=0x5555563eb120) at ../src/openrct2/GameState.cpp:230
No locals.
#5  0x00007ffff6e087a8 in OpenRCT2::GameState::Update (this=0x5555563eb120) at ../src/openrct2/GameState.cpp:127
        i = 0
        numUpdates = 1
#6  0x00007ffff6df279d in OpenRCT2::Context::Update (this=0x555555a351b0) at ../src/openrct2/Context.cpp:965
        currentUpdateTick = 211577121
#7  0x00007ffff6df2586 in OpenRCT2::Context::RunVariableFrame (this=0x555555a351b0) at ../src/openrct2/Context.cpp:920
        currentTick = 211577121
        draw = true
        elapsed = 16
#8  0x00007ffff6df22a3 in OpenRCT2::Context::RunFrame (this=0x555555a351b0) at ../src/openrct2/Context.cpp:855
        useVariableFrame = true
#9  0x00007ffff6df221f in OpenRCT2::Context::RunGameLoop (this=0x555555a351b0) at ../src/openrct2/Context.cpp:830
No locals.
#10 0x00007ffff6df2035 in OpenRCT2::Context::Launch (this=0x555555a351b0) at ../src/openrct2/Context.cpp:806
No locals.
#11 0x00007ffff6df0061 in OpenRCT2::Context::RunOpenRCT2 (this=0x555555a351b0, argc=2, argv=0x7fffffffce38) at ../src/openrct2/Context.cpp:211
No locals.
#12 0x0000555555616dce in main (argc=2, argv=0x7fffffffce38) at ../src/openrct2-ui/Ui.cpp:60
        env = std::shared_ptr<OpenRCT2::IPlatformEnvironment> (use count 8, weak count 0) = {
          get() = 0x55555599e5b0
        }
        audioContext = std::shared_ptr<OpenRCT2::Audio::IAudioContext> (use count 2, weak count 0) = {
          get() = 0x5555559a8bc0
        }
        uiContext = std::shared_ptr<OpenRCT2::Ui::IUiContext> (use count 4, weak count 0) = {
          get() = 0x5555559e04f0
        }
        context = std::unique_ptr<OpenRCT2::IContext> = {
          get() = 0x555555a351b0
        }
        runGame = 1
>>> p *vehicle_entry
$2 = {
  rotation_frame_mask = 0, 
  num_vertical_frames = 0 '\000', 
  num_horizontal_frames = 0 '\000', 
  spacing = 139456, 
  car_mass = 0, 
  tab_height = 0 '\000', 
  num_seats = 18 '\022', 
  sprite_flags = 1, 
  sprite_width = 255 '\377', 
  sprite_height_negative = 255 '\377', 
  sprite_height_positive = 255 '\377', 
  animation = 0 '\000', 
  flags = 67072, 
  base_num_frames = 1, 
  base_image_id = 116866, 
  restraint_image_id = 0, 
  gentle_slope_image_id = 0, 
  steep_slope_image_id = 0, 
  vertical_slope_image_id = 0, 
  diagonal_slope_image_id = 0, 
  banked_image_id = 0, 
  inline_twist_image_id = 0, 
  flat_to_gentle_bank_image_id = 0, 
  diagonal_to_gentle_slope_bank_image_id = 0, 
  gentle_slope_to_bank_image_id = 0, 
  gentle_slope_bank_turn_image_id = 0, 
  flat_bank_to_gentle_slope_image_id = 0, 
  {
    curved_lift_hill_image_id = 0, 
    corkscrew_image_id = 0
  }, 
  no_vehicle_images = 32, 
  no_seating_rows = 0 '\000', 
  spinning_inertia = 0 '\000', 
  spinning_friction = 0 '\000', 
  friction_sound_id = 255 '\377', 
  log_flume_reverser_vehicle_type = 0 '\000', 
  sound_range = 255 '\377', 
  double_sound_frequency = 0 '\000', 
  powered_acceleration = 0 '\000', 
  powered_max_speed = 0 '\000', 
  car_visual = 1 '\001', 
  effect_visual = 1 '\001', 
  draw_order = 6 '\006', 
  num_vertical_frames_override = 0 '\000', 
  peep_loading_waypoint_segments = 0 '\000', 
  pad_62 = "\000\000\000\000\000", 
  peep_loading_waypoints = std::vector of length 0, capacity 0, 
  peep_loading_positions = std::vector of length 0, capacity 0
}
```

**Comment by duncanspumpkin** [comment]
Closed by #9120
